Modules/Core/Common/include/itkImageRandomConstIteratorWithIndex.hxx

```diff
@@ -31,14 +31,15 @@ ImageRandomConstIteratorWithIndex<TImage>::ImageRandomConstIteratorWithIndex(con
 
 template <typename TImage>
 void
-ImageRandomConstIteratorWithIndex<TImage>::SetNumberOfSamples(SizeValueType number)
+ImageRandomConstIteratorWithIndex<TImage>::SetNumberOfSamples(ImageRandomConstIteratorWithIndex::SizeValueType number)
 {
   m_NumberOfSamplesRequested = number;
 }
 
 template <typename TImage>
 auto
-ImageRandomConstIteratorWithIndex<TImage>::GetNumberOfSamples() const -> SizeValueType
+ImageRandomConstIteratorWithIndex<TImage>::GetNumberOfSamples() const
+  -> ImageRandomConstIteratorWithIndex::SizeValueType
```

Comment:
Thanks @fdgStilla, but very unfortunate! It doesn't look very nice to me, to have those added `ImageRandomConstIteratorWithIndex::` prefixes!  What about moving the bodies of the two very little member functions inside the class definition instead, in "itkImageRandomConstIteratorWithIndex.h"?  I think I would like that much better 😃 

BTW, I would suggest the remove the word "FEM" from the message of this specific commit, as the problem appears more general. Still it appears helpful to mention the addressed error message in the body of the commit text:

> error C2244: 'itk::ImageRandomConstIteratorWithIndex<TImage>::SetNumberOfSamples': unable to match function definition to an existing declaration

Reply:
Done, I added a fix for LabelMap but since the method was longer I kept the code in the hxx file, tell me if this is ok

Reply:
Coincidentally I was just thinking on how to shorten the LabelMap method 😸  The `for` loops in the `LabelMap::GetNthLabelObject` overloads could be replaced with something like `return std::next(m_LabelObjectContainer.begin(), pos)->second`. Anyway, that may be beyond the scope of your PR!

So it's ok to me to leave the `LabelMap::GetNthLabelObject` implementation in its hxx file. You may still consider using `Self::`, instead of `LabelMap::`, but that's just a matter of taste, I guess.